lib/metababel/bt2_stream_classes_generator.rb

```diff
@@ -803,4 +813,76 @@ def initialize(parent:, options:, selector_field_class: nil)
       end
     end
   end
+
+  class BTEnvironmentClass
+    include BTPrinter
+    extend BTFromH
+    attr_reader :parent, :entries
+
+    def initialize(parent:, entries: [])
+      @parent = parent
+      @entries = entries.map { |entry| BTValueClass.from_h(self, entry) }
+    end
+
+    def get_getter(event:, arg_variables:)
+      scope do
+        @entries.each do |entry|
+          entry.get_getter(trace: 'trace', arg_variables: arg_variables)
+        end
+      end
+    end
+  end
+
+  class BTValueClass
+    include BTPrinter
+    using HashRefinements
+
+    def self.from_h(parent, model)
+      key = model.delete(:type)
+      raise "No type in #{model}" unless key
+
+      h = { 'string' => BTValueClass::String,
+            'integer_signed' => BTValueClass::IntegerSigned }.freeze
+      raise "Type #{key} not supported" unless h.include?(key)
+      h[key].from_h(parent, model)
+    end
+
+    def get_getter(trace:, arg_variables:)
+      var_name = self.name
+      var_type = self.class.instance_variable_get(:@bt_type)
+
+      var = GeneratedArg.new(var_type, var_name)
+      arg_variables.fetch_append('outputs', var)
+
+      bt_func_get = self.class.instance_variable_get(:@bt_func)
+      pr "const bt_value *#{var_name}_value = bt_trace_borrow_environment_entry_value_by_name_const(#{trace}, \"#{var_name}\");"
+      pr "#{var_name} = #{bt_func_get}(#{var_name}_value);"
+    end
+  end
+
+  class BTValueClass::String < BTValueClass
+    extend BTFromH
+    attr_reader :parent, :name
+
+    @bt_type = 'const char*'
+    @bt_func = 'bt_value_string_get'
+
+    def initialize(parent:, name:)
+      @parent = parent
```

Comment:
No need for parent, as we will never need to recurse back

Reply:
Good point, I only follow the convention, like in the other classes such as BTFieldClass

Reply:
We need the parent since the `from_h`.